src/ceph-volume/ceph_volume/devices/lvm/batch.py

```diff
@@ -98,184 +98,341 @@ def filter_devices(args):
     unused_devices = [device for device in args.devices if not device.used_by_ceph]
     # only data devices, journals can be reused
     used_devices = [device.abspath for device in args.devices if device.used_by_ceph]
-    args.filtered_devices = {}
+    filtered_devices = {}
     if used_devices:
         for device in used_devices:
-            args.filtered_devices[device] = {"reasons": ["Used by ceph as a data device already"]}
+            filtered_devices[device] = {"reasons": ["Used by ceph as a data device already"]}
         logger.info("Ignoring devices already used by ceph: %s" % ", ".join(used_devices))
     if len(unused_devices) == 1:
         last_device = unused_devices[0]
         if not last_device.rotational and last_device.is_lvm_member:
             reason = "Used by ceph as a %s already and there are no devices left for data/block" % (
                 last_device.lvs[0].tags.get("ceph.type"),
             )
-            args.filtered_devices[last_device.abspath] = {"reasons": [reason]}
+            filtered_devices[last_device.abspath] = {"reasons": [reason]}
             logger.info(reason + ": %s" % last_device.abspath)
             unused_devices = []
 
-    return unused_devices
+    return unused_devices, filtered_devices
 
 
 class Batch(object):
 
     help = 'Automatically size devices for multi-OSD provisioning with minimal interaction'
 
     _help = dedent("""
-    Automatically size devices ready for OSD provisioning based on default strategies.
+    Automatically size devices ready for OSD provisioning.
+    The 'auto' subcommand will automtatically determine the strategy (based on
+    detection and presence of the drives rotaional flags.
+
+    The 'bluestore' and 'filestore' subcommands allow to specfify lists of
```

Comment:
why are these subcommands added vs. just using the existing `--filestore` and `--bluestore` flags?

Reply:
Because having the object store as a subcommand allows for proper separation of bluestore (or filestore) specific arguments. I.e. this allows the argument parser to reject, say the use of `--journal-size` with bluestore instead of add a code validation (or simply ignore it).

Reply:
This is not how any other part of ceph-volume currently operates with both objectstores. Not that it means it can't change, but just to say that there is no need for extra validation to be done. In the case of `lvm create` for example, if `--filestore` is specified then any bluestore flag is ignored